daml/Marketplace/Settlement/Hierarchical/Util.daml

```diff
@@ -26,22 +26,26 @@ createInstructions isCash operator agent rootProvider senderAccount receiverAcco
   else if senderAccount.provider == rootProvider then do
     (_, providerSettlementInfo) <- fetchByKey @Custody.SettlementInfo (operator, receiverAccount.provider)
     let providerAccount = getAccount isCash providerSettlementInfo receiverAccount.owner
-    sis1 <- createInstruction receiverAccount.provider receiverAccount.owner (fromSome providerSettlementInfo.ownAccount) receiverAccount (asset with id.signatories = fromList [providerSettlementInfo.party]) instructionIdx
+        assetSi = if isCash then (asset with id.signatories = fromList [providerSettlementInfo.party]) else asset
```

Comment:
This change is made so that the asset.id.signatories does not change and we can use it as key to fetch Claim Token contract for instruments that are not cash. (as bond should be fully backed by bond registrar, I think this makes sense, lemme know if otherwise)
I am not sure why we would like to change the asset.id.signatories for cash (guess something related to CBDC concept : cash provided by cash provider is different from cash provided by lead manager) @georg-da do you mind to explain a bit on this regards? thanks a lot!

Reply:
I think we need to leave that as it was. Reason being that the bond custodial holdings need to have the custodian as provider (ie. signatory), not the bond registrar. That's because they represent holdings "on behalf", ie. liabilities of the custodian, and not actual legal bond ownership (which would be at the bond registrar).

Reply:
We should ensure elsewhere that claim tokens can only be created at the top level (ie. at the bond registrar)

Reply:
As discussed with @GeorgSchneider, since we rely mainly on Account in AssetDeposit to represent the holdings.
It's ok to keep assetDeposit.asset.id.signatories fixed on top level (ie issuer, bond registrar)